src/lib/viz/layer/Layer.ts

```diff
@@ -21,6 +22,16 @@ import { FunctionFilterApplicator } from '../filters/FunctionFilterApplicator';
 import { ColumnFilters } from '../filters/types';
 import { basicStyle } from '../style/helpers/basic-style';
 
+const DATA_READY_EVENT = 'layer:data:ready';
+const DATA_CHANGED_EVENT = 'layer:data:changed';
+
+export interface LayerDataState {
+  isFirstTime: boolean;
+  isPanning: boolean;
+  isZooming: boolean;
+  isRotating: boolean;
```

Comment:
I'm not sure that we want to distinguish these 'subtypes'. We might miss some (for example what happens if we change the pitch?) but in any case, we always treat them together, so I would prefer a common state, named more generically, like 'isViewChanging' or something similar...

Reply:
Just to add some other angle in case it is useful: this is VL approach: https://github.com/CartoDB/carto-vl/blob/v1.4.4/src/Layer.js#L19 simplified to 3 states: initial / updating / idle

Reply:
I think they could be interesting in the future knowing the reason, but know in the current implementation it does not matter

Reply:
Changed to use 1 unique state, so this one does not applies